src/ds/d500/d500-fw-update-device.h

```diff
@@ -16,10 +16,13 @@ namespace librealsense
 
         virtual bool check_fw_compatibility(const std::vector<uint8_t>& image) const override;
         virtual void update(const void* fw_image, int fw_image_size, rs2_update_progress_callback_sptr = nullptr) const override;
-        virtual bool wait_for_manifest_completion(std::shared_ptr<platform::usb_messenger> messenger, const rs2_dfu_state state,
-            std::chrono::seconds timeout_seconds, rs2_update_progress_callback_sptr update_progress_callback) const override;
+        bool wait_for_manifest_completion(std::shared_ptr<platform::usb_messenger> messenger, const rs2_dfu_state state,
+            std::chrono::seconds timeout_seconds, rs2_update_progress_callback_sptr update_progress_callback) const;
+        virtual void dfu_manifest_phase(const platform::rs_usb_messenger& messenger, rs2_update_progress_callback_sptr update_progress_callback) const override;
 
     private:
         std::string parse_serial_number(const std::vector<uint8_t>& buffer) const;
+        void report_progress_and_wait_for_fw_burn(rs2_update_progress_callback_sptr update_progress_callback, int required_dfu_time) const;
+        mutable bool _wait_instead_of_sampling_manifest_reset = false;
```

Comment:
Can we add a comment to why we need this new member?

Reply:
ok